sympy/solvers/solveset.py

```diff
@@ -1117,3 +1118,154 @@ def linsolve(system, *symbols):
     # Return solutions
     solution = FiniteSet(tuple(solution))
     return solution
+
+
+
+###############################################################################
+################################ NLINSOLVE #####################################
+###############################################################################
+
+
+def nlinsolve_matrix(system, symbols):
+    r"""
+    reference : http://people.math.gatech.edu/~aleykin3/math4803spr13/BOOK/chapter1.pdf
+    page :9 example = 2.2
+    currently for 2 equations
+
+    Examples
+    >>> from sympy.solvers.solveset import nlinsolve_matrix
+    >>> nlinsolve_matrix([x*y - 1, 4*x**2 + y**2 - 5],[x,y])
+    >>> {(-1, -1), (-1/2, -2), (1/2, 2), (1, 1)}
+
+    This method will help in this type of cases(nlinsolve calling this method when 
+    solve_poly_system cannot solve non zero dimensional nonlinear system of equations) :
+    >>> nlinsolve([(x + y)**2 - 4, x + y - 2],[x,y])
+    >>> {-y + 2}
```

Comment:
I think the API should be compatible with that of `linsolve`.

With that thought in mind, the ideal output should be
`{(-y + 2, y)}`